src/mon/NVMeofGwMon.cc

```diff
@@ -211,18 +213,37 @@ void NVMeofGwMon::update_from_paxos(bool *need_bootstrap)
 
 void NVMeofGwMon::check_sub(Subscription *sub)
 {
-  dout(10) << "sub->next , map-epoch " << sub->next
-	   << " " << map.epoch << dendl;
-  if (sub->next <= map.epoch)
-  {
-    dout(10) << "Sending map to subscriber " << sub->session->con
-	     << " " << sub->session->con->get_peer_addr() << dendl;
-    sub->session->con->send_message2(make_message<MNVMeofGwMap>(map));
-
-    if (sub->onetime) {
-      mon.session_map.remove_sub(sub);
-    } else {
-      sub->next = map.epoch + 1;
+//  dout(10) << "sub->next , map-epoch " << sub->next   << " " << map.epoch << dendl;
+  for (auto& created_map_pair: map.created_gws) {
+    auto& group_key = created_map_pair.first;
+    NvmeGwMonStates& gw_created_map = created_map_pair.second;
+    for (auto& gw_created_pair: gw_created_map) {
+      auto& gw_id = gw_created_pair.first;
+      if ( (gw_created_pair.second.availability !=
+          gw_availability_t::GW_CREATED ) &&
+         (gw_created_pair.second.addr_vect ==
+          entity_addrvec_t(sub->session->con->get_peer_addr() ))
+         )
+      {
+        dout(10) << "found gw-vect " << gw_created_pair.second.addr_vect
+          << " GW " << gw_id << " group-key " << group_key <<  dendl;
+        dout(10) << "sub->next(epoch) " << sub->next << " map.Gw_epoch "
+           << map.Gw_epoch[group_key].epoch << dendl;
+        if (sub->next <= map.Gw_epoch[group_key].epoch) {
+          dout(4) << "Send unicast map to GW "<< gw_id << dendl;
+          NVMeofGwMap unicast_map;
+          unicast_map.created_gws[group_key][gw_id]
+              = map.created_gws[group_key][gw_id];
+          // respond with a map slice correspondent to the same GW
+          unicast_map.epoch =  map.Gw_epoch[group_key].epoch;//map.epoch;
+          sub->session->con->send_message2(make_message<MNVMeofGwMap>(unicast_map));
+          if (sub->onetime) {
+            mon.session_map.remove_sub(sub);
+          } else {
+            sub->next = map.Gw_epoch[group_key].epoch + 1;
```

Comment:
You only increment next if you need to send it.  How does next change otherwise?